tensorboard/components_polymer3/tf_storage/storage.ts

```diff
@@ -271,7 +271,9 @@ function readComponent(): string {
 function writeComponent(component: string, useLocationReplace = false) {
   if (useHash()) {
     if (useLocationReplace) {
-      window.location.replace('#' + component);
+      const url = new URL(window.location.href);
+      url.hash = component;
+      window.history.replaceState(undefined, undefined, url.toString());
```

Comment:
MDN suggests the first 2 parameters are non-optional.  Should we pass `replaceState(null, '', url.toString())`?

https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState